source/credentials_provider_anonymous.c

```diff
@@ -42,9 +42,6 @@ struct aws_credentials_provider *aws_credentials_provider_new_anonymous(
     struct aws_credentials_provider *provider = aws_mem_calloc(allocator, 1, sizeof(struct aws_credentials_provider));
 
     struct aws_credentials *credentials = aws_credentials_new_anonymous(allocator);
```

Comment:
Probably fine here, but I think it would be better to keep the null check as a design pattern. What if this function starts failing in the future? We should not make assumptions about the return value at this level.